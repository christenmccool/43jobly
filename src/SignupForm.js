import React, {useState} from 'react';
import './Form.css';

const SignupForm = ({signup}) => {
  const initialFormData = {username: "", firstName: "", lastName: "", email: "", password: ""}
  const [formData, setFormData] = useState(initialFormData);

  const handleChange = (evt) => {
    const {name, value} = evt.target;
    setFormData({...formData, [name]: value});
  }

  const handleSubmit = (evt) => {
    evt.preventDefault();
    signup(formData)
    setFormData(initialFormData);
  }

  return (
    <div className="Form">
      <h1 className="Form-title">Sign Up</h1>
      <form className="Form-form" onSubmit={handleSubmit}>
        <label htmlFor="username">Username</label>
        <input 
          id="username" 
          name="username" 
          type="text" 
          value={FormData.username}
          onChange={handleChange}
          required
        />
        <label htmlFor="password">Password</label>
        <input 
          id="password" 
          name="password" 
          type="password" 
          value={FormData.password}
          onChange={handleChange}
          required
        />
        <label htmlFor="firstName">First name</label>
        <input 
          id="firstName" 
          name="firstName" 
          type="text" 
          value={FormData.firstName}
          onChange={handleChange}
          required
        />
        <label htmlFor="lastName">Last name</label>
        <input 
          id="lastName" 
          name="lastName" 
          type="text" 
          value={FormData.lastName}
          onChange={handleChange}
          required
        />
        <label htmlFor="email">Email</label>
        <input 
          id="email" 
          name="email" 
          type="email" 
          value={FormData.email}
          onChange={handleChange}
          required
        />
        <button>Submit</button>
      </form>
    </div>
  )
}

export default SignupForm;