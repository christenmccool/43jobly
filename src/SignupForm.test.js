import { render, asFragment } from '@testing-library/react';
import SignupForm from './SignupForm';

test('if it renders without crashing', () => {
  render(<SignupForm />);
});

test('if it matches snapshot', () => {
  const {asFragment} = render(<SignupForm />);
  expect(asFragment()).toMatchSnapshot();
});