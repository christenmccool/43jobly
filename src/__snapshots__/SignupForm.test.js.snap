// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <h1
      class="Form-title"
    >
      Profile
    </h1>
    <form
      class="Form-form"
    >
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
      <label
        for="firstName"
      >
        First name
      </label>
      <input
        id="firstName"
        name="firstName"
        required=""
        type="text"
        value=""
      />
      <label
        for="lastName"
      >
        Last name
      </label>
      <input
        id="lastName"
        name="lastName"
        required=""
        type="text"
        value=""
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        required=""
        type="email"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
